### ISSUE TITLE
Compilation error with OneAPI icpx and OpenMP backend

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#1040]
Attempting to compile with OneAPI dpc++ compiler resulted in compilation errors in KokkosBatched_Vector_SIMD_Arith.hpp:

```
In file included from /ascldap/users/ndellin/kokkos-kernels/src/batched/KokkosBatched_Vector_SIMD.hpp:896:
/ascldap/users/ndellin/kokkos-kernels/src/batched/KokkosBatched_Vector_SIMD_Arith.hpp:69:1: error: expected a for, while, or do-while loop to follow '#pragma ivdep'
#pragma omp simd
...
```

Based on the error, I'm assuming the guards for the `KOKKOSKERNELS_ENABLE_OMP_SIMD` macro may need to be refined? https://github.com/kokkos/kokkos-kernels/blob/ae9b047cf5a98f150c7b85158a3592dd0b874750/src/KokkosKernels_Macros.hpp#L58-L78

Compiler version:
```
bash-4.2$ icpx --version
Intel(R) oneAPI DPC++ Compiler 2021.1 (2020.10.0.1113)
```

Reproducer (Blake testbed):
```
module load cmake/3.19.3 intel/oneAPI/base-toolkit/2021.1.1

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=OpenMP --arch=SKX --compiler=icpx --cxxflags="-O3  " --cxxstandard="14" --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft --no-examples
```

**Comment by lucbv** [comment]
Was Blake's OS upgraded? That would be a good news : )

**Comment by nmhamster** [comment]
Blake is running RHEL 7.9.

**Comment by lucbv** [comment]
Ah, so we can build but not run, still good to be able to test the build.

@ndellingwood are building with the SYCL backend?

**Comment by ndellingwood** [comment]
@lucbv no, this was with OpenMP

**Comment by ndellingwood** [comment]
Looks like this should have been fixed by #1040, marking InDevelop
